content/blog/2017/2017-01-01-conditional.adoc

```diff
@@ -0,0 +1,309 @@
+---
+layout: post
+title: "Migrating to Pipeline: Conditional BuildStep plugin"
+tags:
+- pipeline
+- freestyle
+- plugins
+- conditional-build-step
+author: lnewman
+---
+
+NOTE: This is a guest post by link:https://github.com/bitwiseman[Liam Newman],
+Technical Evangelist at link:http://cloudbees.com[CloudBees].
+
+The
+link:https://wiki.jenkins-ci.org/display/JENKINS/Conditional+BuildStep+Plugin[Conditional BuildStep plugin]
+is a powerful tool that has allowed Jenkins users to write Jenkins jobs with
+complex conditional logic.  The plugin also includes a set of
+token macros that expose a wide range of job state information on which to base
+those conditions.  Conceptually, mapping the conditional behavior to
+Pipeline is easy, but actually migrating full jobs will be more involved.
+
+== Introduction
+
+There are number of plugins, some that have been around since before Jenkins was
+Jenkins, that enable users to create pipelines in Jenkins.
+
+With all the new developments in Jenkins Pipeline and with Declarative Pipeline
+on the horizon, it's easy forget what we did to create pipelines before Pipeline.
+
+In particular, creating a pipeline with conditional behavior was one of the harder things to
+do in Jenkins.
+
+Basic job chaining worked well in most cases, but what about when
+a pipeline needed conditional logic?
+
+
+// It required job chaining, parameterized job triggering, and loading properties from shell scripts
```

Comment:
I should have left an accompanying `// blah blah - I'll come back to this.` comment.